### PR TITLE
Feature/items pagination

### PR DESCRIPTION
**Pull request by mmiranda-stackit**

### CODE REVIEW

File: src/common/middlewares/pager.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { NextFunction } from 'express';
+import { constants } from '../constants';
+
+@Injectable()
+export class PagerMiddleware implements NestMiddleware {
+  use(req: any, _: any, next: NextFunction) {
+    const { pagination } = constants;
+    req.query.limit = +req.query.limit || pagination.defaultLimit;
+    req.query.page = +req.query.page || pagination.defaultPage;
+    req.query.limit =
```
Comment (juliswer) [inline src/common/middlewares/pager.middleware.ts:11]: Why are you setting twice the req.query.limit?

File: src/items/controllers/item.controller.ts [hunk 2]
```diff
@@ -5,31 +5,26 @@ import { BuyRequestDto } from '../dto/buy-request.dto';
...
-  search(@Query() { topic }: SearchRequestDto) {
-    const notTopicMessage = 'Please provide a topic to search';
-    if (!topic) throw new NotFoundException(notTopicMessage);
-    const topicValue: SearchRequestDto = {
-      topic,
-    };
-    return this.itemService.search(topicValue);
+  @Get('/paginate')
+  paginate(@Query() paginationDto: ItemPaginationDto) {
+    console.log('paginationDto', paginationDto);
```
Comment (juliswer) [inline src/items/controllers/item.controller.ts]: remove console.log

**Review by juliswer — COMMENTED** [review]
Looks good. Please check my comments and resolve them before merging (if they are already fixed, just merge main carelessly)

**Review by juliswer — APPROVED** [review]

File: src/common/middlewares/pager.middleware.ts (hunk 1 above)
Comment (mmiranda-stackit) [reply src/common/middlewares/pager.middleware.ts:11]: The first one is making sure the limit has a value. The second one is making sure that value is not bigger than the max default (Imagine you send a limit of 1,000,000), The first one just passes through bcuz it is defined, but the second one delimitates the value to the max configured.

File: src/items/controllers/item.controller.ts (hunk 2 above)
Comment (mmiranda-stackit) [reply src/items/controllers/item.controller.ts]: You caught me

Comment (mmiranda-stackit) [reply src/items/controllers/item.controller.ts]: Are you sure it is the last commit you are reviewing ?

**Review by GimerStackIt — APPROVED** [review]

**Pull request merged**